src/Core/Form/IdentifiableObject/DataHandler/RootCategoryFormDataHandler.php

```diff
@@ -26,101 +26,24 @@
 use PrestaShop\PrestaShop\Core\Domain\Category\Command\AddRootCategoryCommand;
 use PrestaShop\PrestaShop\Core\Domain\Category\Command\EditRootCategoryCommand;
-use PrestaShop\PrestaShop\Core\Domain\Category\ValueObject\CategoryId;
-use PrestaShop\PrestaShop\Core\Image\Uploader\ImageUploaderInterface;
-use Symfony\Component\HttpFoundation\File\UploadedFile;
 
 /**
  * Creates/updates root category from data submited in category form
  */
-final class RootCategoryFormDataHandler implements FormDataHandlerInterface
+final class RootCategoryFormDataHandler extends AbstractCategoryFormDataHandler
 {
-    /**
-     * @var CommandBusInterface
-     */
-    private $commandBus;
-
-    /**
-     * @var ImageUploaderInterface
-     */
-    private $categoryCoverUploader;
-
-    /**
-     * @var ImageUploaderInterface
-     */
-    private $categoryThumbnailUploader;
-
-    /**
-     * @var ImageUploaderInterface
-     */
-    private $categoryMenuThumbnailUploader;
-
-    /**
-     * @param CommandBusInterface $commandBus
-     * @param ImageUploaderInterface $categoryCoverUploader
-     * @param ImageUploaderInterface $categoryThumbnailUploader
-     * @param ImageUploaderInterface $categoryMenuThumbnailUploader
-     */
-    public function __construct(
-        CommandBusInterface $commandBus,
-        ImageUploaderInterface $categoryCoverUploader,
-        ImageUploaderInterface $categoryThumbnailUploader,
-        ImageUploaderInterface $categoryMenuThumbnailUploader
-    ) {
-        $this->commandBus = $commandBus;
-        $this->categoryCoverUploader = $categoryCoverUploader;
-        $this->categoryThumbnailUploader = $categoryThumbnailUploader;
-        $this->categoryMenuThumbnailUploader = $categoryMenuThumbnailUploader;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function create(array $data)
-    {
-        $command = $this->createAddRootCategoryCommand($data);
-
-        /** @var CategoryId $categoryId */
-        $categoryId = $this->commandBus->handle($command);
-
-        $this->uploadImages(
-            $categoryId->getValue(),
-            $data['cover_image'],
-            $data['thumbnail_image'],
-            $data['menu_thumbnail_images']
-        );
-
-        return $categoryId->getValue();
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function update($rootCategoryId, array $data)
-    {
-        $command = $this->createEditRootCategoryCommand($rootCategoryId, $data);
-
-        $this->commandBus->handle($command);
-
-        $this->uploadImages(
-            $rootCategoryId,
-            $data['cover_image'],
-            $data['thumbnail_image'],
-            $data['menu_thumbnail_images']
-        );
-    }
-
     /**
      * Creates command with form data for adding new root category
      *
      * @param array $data
      *
      * @return AddRootCategoryCommand
      */
-    public function createAddRootCategoryCommand(array $data)
```

Comment:
This is a BC breaks, you have to document it :thinking:

Reply:
Updated PR desecription. Or do I need to add comment to code as well?

Reply:
It's indeed a BC break so it should documented But this method should have never been public in the first place